core/src/main/scala/org/broadinstitute/dsde/rawls/fastpass/FastPassService.scala

```diff
@@ -0,0 +1,444 @@
+import org.broadinstitute.dsde.workbench.openTelemetry.OpenTelemetryMetrics
+
+import scala.concurrent.{ExecutionContext, Future}
+
+object FastPassService {
+  def constructor(config: FastPassConfig,
+                  googleIamDao: GoogleIamDAO,
+                  googleStorageDAO: GoogleStorageDAO,
+                  samDAO: SamDAO,
+                  terraBillingProjectOwnerRole: String,
+                  terraWorkspaceCanComputeRole: String,
+                  terraWorkspaceNextflowRole: String,
+                  terraBucketReaderRole: String,
+                  terraBucketWriterRole: String
+  )(ctx: RawlsRequestContext, dataAccess: DataAccess)(implicit
+    executionContext: ExecutionContext,
+    openTelemetry: OpenTelemetryMetrics[IO]
+  ): FastPassService =
+    new FastPassService(
+      ctx,
+      dataAccess,
+      config,
+      googleIamDao,
+      googleStorageDAO,
+      samDAO,
+      terraBillingProjectOwnerRole,
+      terraWorkspaceCanComputeRole,
+      terraWorkspaceNextflowRole,
+      terraBucketReaderRole,
+      terraBucketWriterRole
+    )
+
+  val policyBindingsQuotaLimit = 1500
+  val maxProjectRoleBindingsPerUser = 6 // 3 possible roles for each user and pet service account
+  val maxBucketRoleBindingsPerUser = 2 // 1 possible role for each user and pet service account
+
+}
+
+class FastPassService(protected val ctx: RawlsRequestContext,
+                      protected val dataAccess: DataAccess,
+                      protected val config: FastPassConfig,
+                      protected val googleIamDao: GoogleIamDAO,
+                      protected val googleStorageDAO: GoogleStorageDAO,
+                      protected val samDAO: SamDAO,
+                      protected val terraBillingProjectOwnerRole: String,
+                      protected val terraWorkspaceCanComputeRole: String,
+                      protected val terraWorkspaceNextflowRole: String,
+                      protected val terraBucketReaderRole: String,
+                      protected val terraBucketWriterRole: String
+)(implicit protected val executionContext: ExecutionContext, val openTelemetry: OpenTelemetryMetrics[IO])
+    extends LazyLogging {
+
+  private val openTelemetryTags: Map[String, String] = Map("service" -> "FastPassService")
+  private def samWorkspaceRoleToGoogleProjectIamRoles(samResourceRole: SamResourceRole) =
+    samResourceRole match {
+      case SamWorkspaceRoles.projectOwner =>
+        Set(terraBillingProjectOwnerRole, terraWorkspaceCanComputeRole, terraWorkspaceNextflowRole)
+      case SamWorkspaceRoles.owner      => Set(terraWorkspaceCanComputeRole, terraWorkspaceNextflowRole)
+      case SamWorkspaceRoles.canCompute => Set(terraWorkspaceCanComputeRole, terraWorkspaceNextflowRole)
+      case _                            => Set.empty[String]
+    }
+
+  private def samWorkspaceRolesToGoogleBucketIamRoles(samResourceRole: SamResourceRole) =
```

Comment:
Can you use `WorkspaceService.getGoogleBucketPermissionsFromRoles`?

Reply:
we are planning on ripping the FastPassService class out of rawls so we want to keep it as independent as possible